FerramAerospaceResearch/FARAeroComponents/VehicleOcclusion.cs

```diff
@@ -0,0 +1,828 @@
+        {
+            PassStarted = false;
+            if (!OnValidPhysicsVessel)
+                return;
+            if (state == State.Running)
+                HandleJobCompletion(raycastJobTracker, FixedUpdateMetric);
+        }
+
+        public void UpdateEarly()
+        {
+            if (!OnValidPhysicsVessel)
+                return;
+            if (state == State.Running)
+            {
+                int threadCount = System.Environment.ProcessorCount - 1;
+                int availableJobs = math.max(OcclusionSettings.MaxJobs - JobsInCurrentPass, 0);
+                int updateJobs = math.min(availableJobs, math.max(1, Settings.OcclusionSettings.JobsPerThread * threadCount));
+                SetupDefaultDirs(directionList, Vessel);
+                LaunchJobs(Vessel, directionPreprocessInfos, directionList, occlusionPointsList, updateJobs);
+                if (!PassStarted)
+                {
+                    PassStarted = true;
+                    JobsInCurrentPass = 0;
+                }
+                JobsInCurrentPass += jobsInProgress;
+            }
+        }
+
+        public void LateUpdateComplete()
+        {
+            PassStarted = false;
+            if (!OnValidPhysicsVessel)
+                return;
+            if (state == State.Running && jobsInProgress == 0 && processedQuaternionsMap.Length == Quaternions.Length)
+                state = State.Completed;
+            else if (state == State.Running)
+                HandleJobCompletion(raycastJobTracker, UpdateMetric);
+            else if (state == State.Completed)
+            {
+                if (!resetCoroutineRunning)
+                    StartCoroutine(resetWaitCoroutine = WaitForResetCR(OcclusionSettings.ResetInterval));
+            }
+            if (RequestReset)
+                ResetCalculations();
+        }
+
+        IEnumerator WaitForResetCR(float interval)
+        {
+            resetCoroutineRunning = true;
+            yield return new WaitForSeconds(interval);
+            resetCoroutineRunning = false;
+            ResetCalculations();
+        }
+
+        public void HandleJobCompletion(RaycastJobInfo[] tracker, string metric)
+        {
+            JobCompleteWatch.Reset();
+            for (int i=0; i<jobsInProgress; i++)
+            {
+                JobCompleteWatch.Start();
+                tracker[i].processorJob.Complete();
+                JobCompleteWatch.Stop();
+                ProcessRaycastResults(ref tracker[i], allHitSizeMaps[i], allHits[i], allSummedHitAreas[i]);
+            }
+            metrics.AddMeasurement(metric, JobCompleteWatch.Elapsed.TotalMilliseconds);
+            FARLogger.Debug($"Occlusion {metric}: {JobCompleteWatch.Elapsed.TotalMilliseconds:F2}ms");
+        }
+
+        // http://extremelearning.com.au/how-to-evenly-distribute-points-on-a-sphere-more-effectively-than-the-canonical-fibonacci-lattice/
+        private float Lattice_epsilon(int n) => 0.36f;
```

Comment:
Use `const float` if this will not change at runtime

Reply:
Implemented in 0c9479b
(Normally this value should be set based on the lattice size, depending on how you want to tweak the distribution.  For the distribution we want here, though, a single value is correct for basically all sizes.)